Source/Core/DolphinQt2/Settings/AdvancedPane.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Settings/AdvancedPane.h"
+
+#include <QCheckBox>
+#include <QDateTimeEdit>
+#include <QGroupBox>
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QSlider>
+#include <QVBoxLayout>
+
+#include "Core/ConfigManager.h"
+#include "Core/HW/SystemTimers.h"
+
+AdvancedPane::AdvancedPane(QWidget* parent) : QWidget(parent)
+{
+  CreateLayout();
+  ConnectLayout();
+}
+
+void AdvancedPane::ConnectLayout()
+{
+  m_cpu_clock_override_checkbox->setChecked(SConfig::GetInstance().m_OCEnable);
+  connect(m_cpu_clock_override_checkbox, &QCheckBox::toggled, [=](bool enable_cpu_clock_override) {
+    SConfig::GetInstance().m_OCEnable = enable_cpu_clock_override;
+    RefreshEnabled();
+  });
+
+  m_cpu_clock_override_slider->setValue(
+      static_cast<int>(std::ceil(std::log2f(SConfig::GetInstance().m_OCFactor) * 25.f + 100.f)));
+  connect(m_cpu_clock_override_slider, &QSlider::valueChanged, [=](int oc_factor) {
+    // Vaguely exponential scaling?
+    SConfig::GetInstance().m_OCFactor = std::exp2f((oc_factor - 100.f) / 25.f);
+    UpdateCPUClockOverrideSliderLabel();
+  });
+  UpdateCPUClockOverrideSliderLabel();
+
+  m_custom_rtc_checkbox->setChecked(SConfig::GetInstance().bEnableCustomRTC);
+  connect(m_custom_rtc_checkbox, &QCheckBox::toggled, [=](bool enable_custom_rtc) {
+    SConfig::GetInstance().bEnableCustomRTC = enable_custom_rtc;
+    RefreshEnabled();
+  });
+
+  QDateTime initial_date_time;
+  initial_date_time.setTime_t(SConfig::GetInstance().m_customRTCValue);
+  m_custom_rtc_datetime->setDateTime(initial_date_time);
+  connect(m_custom_rtc_datetime, &QDateTimeEdit::dateTimeChanged, [=](const QDateTime& datetime) {
+    SConfig::GetInstance().m_customRTCValue = datetime.toTime_t();
+  });
+
+  RefreshEnabled();
+}
+
+void AdvancedPane::UpdateCPUClockOverrideSliderLabel()
+{
+  int core_clock = SystemTimers::GetTicksPerSecond() / pow(10, 6);
```

Comment:
is the `std::` namespace qualifier intended to be dropped here unlike all the other math functions used?